### ISSUE TITLE
ClosedByInterruptedException not handled in MappedBytes#realCapacity

### ISSUE THREAD

**Issue by bdelbosc** [OpenHFT/Chronicle-Bytes#44]
Hi,
In [`net.openhft.chronicle.bytes.MappedBytes#realCapacity` v1.9.18](https://github.com/OpenHFT/Chronicle-Bytes/blob/chronicle-bytes-1.9.18/src/main/java/net/openhft/chronicle/bytes/MappedBytes.java#L147)  `ClosedByInterruptedException` is catched and ignored. This cause problem with the table directory listing used in Chronicle Queue in case of thread interruption, where there is a `WARN` followed by an `ERROR`, it looks like the original `InterruptedException` is lost.

```
WARN  [Slf4jExceptionHandler$2] Unable to obtain the real size for /tmp/junit6367516383187162035/junit3970794415465406300/killConsumers/offset-default/directory-listing.cq4t
net.openhft.chronicle.core.io.IORuntimeException: java.nio.channels.ClosedByInterruptException
	at net.openhft.chronicle.bytes.MappedFile.actualSize(MappedFile.java:418)
	at net.openhft.chronicle.bytes.MappedBytes.realCapacity(MappedBytes.java:145)
	at net.openhft.chronicle.queue.impl.table.SingleTableStore.acquireValueFor(SingleTableStore.java:203)
	at net.openhft.chronicle.queue.impl.single.TableDirectoryListing.lambda$init$0(TableDirectoryListing.java:53)
	at net.openhft.chronicle.queue.impl.table.SingleTableStore.doWithExclusiveLock(SingleTableStore.java:246)
	at net.openhft.chronicle.queue.impl.single.TableDirectoryListing.init(TableDirectoryListing.java:52)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue.<init>(SingleChronicleQueue.java:141)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder.build(SingleChronicleQueueBuilder.java:188)
       ...
Caused by: java.nio.channels.ClosedByInterruptException
	at java.nio.channels.spi.AbstractInterruptibleChannel.end(AbstractInterruptibleChannel.java:202)
	at sun.nio.ch.FileChannelImpl.size(FileChannelImpl.java:315)
	at net.openhft.chronicle.bytes.MappedFile.actualSize(MappedFile.java:415)

ERROR [AbstractCallablePool] Exception catch in pool: readCheckOffset0 failed. Offset: 0 + adding: 112 > limit: 0 (given: false)
net.openhft.chronicle.bytes.util.DecoratedBufferUnderflowException: readCheckOffset0 failed. Offset: 0 + adding: 112 > limit: 0 (given: false)
	at net.openhft.chronicle.bytes.AbstractBytes.readCheckOffset0(AbstractBytes.java:676)
	at net.openhft.chronicle.bytes.AbstractBytes.readCheckOffset(AbstractBytes.java:661)
	at net.openhft.chronicle.bytes.MappedBytes.readCheckOffset(MappedBytes.java:186)
	at net.openhft.chronicle.bytes.AbstractBytes.readOffsetPositionMoved(AbstractBytes.java:421)
	at net.openhft.chronicle.bytes.AbstractBytes.readSkip(AbstractBytes.java:249)
	at net.openhft.chronicle.bytes.AbstractBytes.readSkip(AbstractBytes.java:32)
	at net.openhft.chronicle.wire.AbstractWire.readDataHeader(AbstractWire.java:201)
	at net.openhft.chronicle.wire.WireIn.readDataHeader(WireIn.java:183)
	at net.openhft.chronicle.queue.impl.table.SingleTableStore.acquireValueFor(SingleTableStore.java:204)
	at net.openhft.chronicle.queue.impl.single.TableDirectoryListing.lambda$init$0(TableDirectoryListing.java:53)
	at net.openhft.chronicle.queue.impl.table.SingleTableStore.doWithExclusiveLock(SingleTableStore.java:246)
	at net.openhft.chronicle.queue.impl.single.TableDirectoryListing.init(TableDirectoryListing.java:52)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue.<init>(SingleChronicleQueue.java:141)
	at net.openhft.chronicle.queue.impl.single.SingleChronicleQueueBuilder.build(SingleChronicleQueueBuilder.java:188)
 ```
I will try to create a unit test.
Regards
ben

**Comment by peter-lawrey** [comment]
I have changed the behaviour to throw an IllegalStateException as the thread shouldn't be running (or creating queues) once interrupted.

**Comment by bdelbosc** [comment]
:+1:

**Comment by hft-team-city** [comment]
Released in [Chronicle-Bytes-2.20.101](https://github.com/OpenHFT/Chronicle-Bytes/releases/chronicle-bytes-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
